core/DataTable/Filter/PivotByDimension.php

```diff
@@ -267,8 +269,12 @@ private function getIntersectedTable(DataTable $table, Row $row)
         }
 
         if ($this->isFetchingBySegmentEnabled) {
-            $segmentValue = $row->getColumn('label');
-            return $this->fetchIntersectedWithThisBySegment($table, $segmentValue);
+            $segment = $row->getMetadata('segment');
```

Comment:
does this fix a different bug? FYI: The segment metadata might not always be set for all reports etc...

Reply:
I see in the description mentions something around fixing segments. Not sure what the problem is and if this fixes it.

Reply:
Yes, saw it while I was working. There was already a skipped test for it. This is why we call the filter above.

Reply:
Can't quite remember, but I think the label fallback is still there?

Reply:
I think I misread your question, but the label fallback is still there (it's in a `?:` expression), in case the segmentValue isn't there. I'll clear up the code since that wasn't obvious.